### ISSUE TITLE
Better error message if a hook fails

### ISSUE THREAD

**Issue by htgoebel** [pyinstaller/pyinstaller#4537]
If a hook failes, you might get tracebacks like this:
```
  File "…\site-packages\PyInstaller\hooks\hook-pendulum.py", line 16, in <module>
    datas = collect_data_files("pendulum.locales", include_py_files=True)
  File "…\site-packages\PyInstaller\utils\hooks\__init__.py", line 721, in collect_data_files
    pkg_base, pkg_dir = get_package_paths(package)
  File "…\site-packages\PyIpulnstaller\utils\hooks\__init__.py", line 535, in get_package_paths
    file_attr = get_module_file_attribute(package)
  File "…\site-packages\PyInstaller\utils\hooks\__init__.py", line 337, in get_module_file_attribute
    raise ImportError
ImportError
```
`ModuleHook` should catch such errors and raise another exception instead. Depending on the error, it should terminate the program (raise SystemExit).

Please are e.g. [`ModuleHook_load_hook_module()`](https://github.com/pyinstaller/pyinstaller/blob/4cad737d27895e0478cb55ad421415278ce2f58f/PyInstaller/depend/imphook.py#L381-382), `ModuleHook._process_hook_func()`.

In he example above the full massage could be:

> Failed to import module {modulename} required by hook for module {hookmodulename}.
Please check whether module {modulename} actually exists and whether the hook is compatible with your version of {hookmodulename}: You might want to read more about hooks in the manual <link> and provide a pull-request to improve PyInstaller.

Same for pre-find-module-path hooks and pre-safe-import-module hooks.

When impementing this, please put new exceptions into `PyInstaller.exceptions`.

**Comment by RS357** [comment]
Hi, could you please assign this bug to me?

**Comment by Legorooj** [comment]
@RS357 can you submit a PR?

**Comment by RS357** [comment]
Hi @Legorooj - please see my PR at https://github.com/pyinstaller/pyinstaller/pull/5077

**Comment by Legorooj** [comment]
Fixed in #5077
